docs/diagrams/CompanyDetailPanelActivityDiagram.puml

```diff
@@ -0,0 +1,27 @@
+@startuml
+skin rose
+skinparam ActivityFontSize 15
+skinparam ArrowFontSize 12
+start
+:User executes command;
+
+'Since the beta syntax does not support placing the condition outside the
+'diamond we place it as the true branch instead.
+
+if () then ([view, edit or add command])
+    :setCurrentViewedCompany(Company company);
+    :clear UniqueCompanyList
+    and add the company to
+    UniqueCompanyList;
+    :company detail panel displays company;
+else ([delete command])
+    :checkDelete(Company company);
+    if () then ([viewing the company to be deleted])
+        :clear UniqueCompanyList;
+        :company detail panel cleared;
+    else ([not viewing the company to be deleted])
```

Comment:
for the else condition, can just put "else" instead of "not viewing the company to be deleted" if there is only one other control path

Reply:
done